docs/jemalloc_usage.md

```diff
@@ -0,0 +1,29 @@
+# JEMalloc usage
+FEX-Emu uses two different jemalloc heap allocators at once.
+- jemalloc: The primary heap allocator
+- jemalloc_glibc: The second heap allocator
+
+## jemalloc - primary heap allocator
+This heap allocator is the primary allocator that FEX-Emu uses. All of FEXCore's `fextl::` namespaced objects will allocate memory through this
+allocator. This allocator has its usage of mmap and munmap overridden to use FEXCore's internal VMA region allocator.
+
+### FEXCore internal VMA region allocator
+- When running a 64-bit guest application, this VMA region allocator is disabled and passes through to mmap and munmap in the host kernel.
+- This only gets interesting when running a 32-bit guest application.
+
+When running a 32-bit guest application, the VMA region allocator will allocate all memory above 4GB virtual address space (Up to 256TB on 48-bit VA
+systems!). It then provides a mmap and munmap interface to allow allocating memory out of this reserved virtual address space. This ensures that all
+of FEX's allocations stay out of the lower 32-bit 4GB VA space, otherwise games would quickly run out of virtual address space.
+
+## jemalloc_glibc - secondary heap allocator
+This heap allocator replaces glibc's allocator using weak symbol overriding. This explicitly does **not** use the FEXCore VMA region allocator. This
+ensures that all allocations that go through this heap allocator will use the kernel mmap and munmap interface. All shared libraries will use this
+allocator including the host-side of thunks.
+
+### Thunks
+When thunks allocate memory, some pointers need to be tracked from which heap allocator they come from when getting freed. In this case some pointers
+might be coming from the guest heap-allocator, and some might be coming from the host heap-allocator.
+
+jemalloc provides a `is_known_allocation` function that allows us to query if the pointer comes from the host heap-allocator. Which FEXCore can use to
+determine which heap should free the pointer. glibc doesn't provide this same interface which is why we need to use jemalloc for this.
```

Comment:
```suggestion
Thunks may allocate memory either through the guest-side (on the guest glibc heap) or the host-side (on the `jemalloc_glibc` heap). To properly free this memory, FEX must be able to determine which heap allocator it belongs to.

`glibc` provides no public interface to do this, but FEX's `jemalloc` fork does. The `is_known_allocation` function is used by FEXCore to query whether a given pointer originated from the `jemalloc_glibc` allocator. This enables FEXCore to determine the appropriate heap for freeing the pointer.
```

(`verify`->`query`, since the former suggests the condition is expected to hold true)

Reply:
Changed phrasing to yours.